File: HERENCIA/EJERCICIO1/Persona.py
class Persona:
    def __init__(self, nombre, apellido, ci):
        self.nombre = nombre
        self.apellido = apellido
        self. ci = ci
    
    def __str__(self):
        return f"Nombre: {self.nombre}, Apellido: {self.apellido}, ci: {self.ci}"
    
class Cliente(Persona):
    def __init__(self, nombre, apellido, ci, nroCompra, idCliente):
        super().__init__(nombre, apellido, ci)
        self.nroCompra = nroCompra
        self.idCliente = idCliente
        
    def __str__(self):
        return f"Nombre: {self.nombre}, Apellido: {self.apellido}, ci: {self.ci}, NroCompra: {self.nroCompra}, ID: {self.idCliente}"
    

class Jefe(Persona):
    def __init__(self, nombre, apellido, ci, sucursal, tipo):
        super().__init__(nombre, apellido, ci)
        self.sucursal = sucursal
        self.tipo = tipo
        
    def __str__(self):
        return f"Nombre: {self.nombre}, Apellido: {self.apellido}, ci: {self.ci}, sucursal: {self.sucursal}, tipo: {self.tipo}"
    

pe = Persona("juan", "perez", 1234566)
print(pe)

cl = Cliente("pedro","picapiedra", 12312312, 23, 191991)
print(cl)

je = Jefe("pablo", "perez", 8776655, "Sucursal Centro", "galeria")
print(je)

File: HERENCIA/EJERCICIO1/Persona.java
package EJERCICIO1;
public class Persona {
    protected String nombre;
    protected String apellido;
    protected int ci;

    public Persona(String nombre, String apellido, int ci) {
        this.nombre = nombre;
        this.apellido = apellido;
        this.ci = ci;
    }

    @Override
    public String toString() {
        return "Nombre: " + nombre + ", Apellido: " + apellido + ", ci: " + ci;
    }
}

class Cliente extends Persona {
    private int nroCompra;
    private int idCliente;

    public Cliente(String nombre, String apellido, int ci, int nroCompra, int idCliente) {
        super(nombre, apellido, ci);
        this.nroCompra = nroCompra;
        this.idCliente = idCliente;
    }

    @Override
    public String toString() {
        return "Nombre: " + nombre + ", Apellido: " + apellido + ", ci: " + ci +
               ", NroCompra: " + nroCompra + ", ID: " + idCliente;
    }
}

class Jefe extends Persona {
    private String sucursal;
    private String tipo;

    public Jefe(String nombre, String apellido, int ci, String sucursal, String tipo) {
        super(nombre, apellido, ci);
        this.sucursal = sucursal;
        this.tipo = tipo;
    }

    @Override
    public String toString() {
        return "Nombre: " + nombre + ", Apellido: " + apellido + ", ci: " + ci +
               ", sucursal: " + sucursal + ", tipo: " + tipo;
    }
}

